proto/gcpb.proto

```diff
@@ -0,0 +1,100 @@
+syntax = "proto3";
+package gcpb;
+
+import "pdpb.proto";
+
+import "gogoproto/gogo.proto";
+import "rustproto.proto";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (rustproto.lite_runtime_all) = true;
+
+option java_package = "org.tikv.kvproto";
+
+service GC {
+  rpc GetAllServiceGroups(GetAllServiceGroupsRequest) returns (GetAllServiceGroupsResponse) {}
+
+  rpc GetMinServiceSafePointByServiceGroup(GetMinServiceSafePointByServiceGroupRequest) returns (GetMinServiceSafePointByServiceGroupResponse) {}
+
+  rpc UpdateGCSafePointByServiceGroup(UpdateGCSafePointByServiceGroupRequest) returns (UpdateGCSafePointByServiceGroupResponse) {}
+
+  rpc UpdateServiceSafePointByServiceGroup(UpdateServiceSafePointByServiceGroupRequest) returns (UpdateServiceSafePointByServiceGroupResponse) {}
+
+  rpc GetAllServiceGroupGCSafePoints(GetAllServiceGroupGCSafePointsRequest) returns (GetAllServiceGroupGCSafePointsResponse) {}
+}
+
+message GetAllServiceGroupsRequest {
+  pdpb.RequestHeader header = 1;
```

Comment:
It would better to be independent to PD's `RequestHeader` & `ResponseHeader`:
1. To make GC service possible to be totally independent to PD (e.g, a standalone micro service in Cloud).
2. Define error type of PD & GC service together seems not reasonable.

Reply:
Should the same separation be done on `PD` side? 
Say, move gc methods outside `grpc_service.go`? And use separate `type GcServer` to provide gc functionalities instead of `GrpcServer`? https://github.com/tikv/pd/blob/6ab2e4448c331de6dbedb61dbec5113e1a2a4122/server/grpc_service.go#L72
What about client? Is a separate `gc client` needed along side PD client?

Reply:
I think yes, and we can try. Unless it is too expensive to do so. @nolouch How do you think ?

Reply:
LGTM, PD client may include `gc client`?